src/auto-complete.directive.ts

```diff
@@ -129,8 +131,11 @@ export class NguiAutoCompleteDirective implements OnInit {
   }
 
   ngOnChanges(changes: SimpleChanges): void {
-    if (changes['ngModel']) {
-      this.ngModel = this.setToStringFunction(changes['ngModel'].currentValue);
+    // if (changes['ngModel']) {
+    //   this.ngModel = this.setToStringFunction(changes['ngModel'].currentValue);
+    // }
+    if (changes["source"]) {
+      this.sourceChanges.emit(this.source);
```

Comment:
This class wasn't implementing the OnChanges lifecycle then those lines weren't fired, so in order to not break anything they were commented out.